lib/app/models/testing_instance.rb

```diff
@@ -99,38 +97,39 @@ def waiting_on_sequence
         sequence_results.first(result: 'wait')
       end
 
-      def final_result
+      def all_passed?
         required_sequences = Inferno::Sequence::SequenceBase.subclasses.reject(&:optional?)
 
-        all_passed = required_sequences.all? do |sequence|
+        required_sequences.all? do |sequence|
           latest_results[sequence.name].try(:result) == 'pass'
         end
+      end
 
-        if all_passed
-          return 'pass'
-        else
-          return 'fail'
-        end
+      def final_result
+        all_passed? ? 'pass' : 'fail'
       end
 
       def fhir_version
         self.module.fhir_version
       end
 
       def module
-        Inferno::Module.get(selected_module)
+        @module ||= Inferno::Module.get(selected_module)
       end
 
       def patient_id
-        resource_references.select { |ref| ref.resource_type == 'Patient' }.first.try(:resource_id)
+        resource_references
+          .select { |ref| ref.resource_type == 'Patient' }
+          .first&.resource_id
       end
 
       def patient_id=(patient_id)
         return if patient_id.to_s == self.patient_id.to_s
 
-        resource_references.select { |ref| ref.resource_type == 'Patient' }
-        # Use destroy directly (instead of on each, so we don't have to reload)
-        resource_references.destroy
+        resource_references
+          .select { |ref| ref.resource_type == 'Patient' }
+          .destroy # Use destroy directly (instead of on each, so we don't have to reload)
```

Comment:
Hmmm, was this a bug before, or did `select` mutate `resource_references`?  Seems like a bug.

Reply:
Oops, I didn't notice that I changed the behavior here. Do you know whether the intent is to delete all resources when the patient changes?